File: unsorted/Vasya_Clerk.go
package kata

func Tickets(peopleInLine []int) string {
	// number of bills in the wallett
	wallet100 := 0
	wallet50 := 0
	wallet25 := 0
	for _, bill := range peopleInLine {
		switch {
		// First: ignore the limited number of 25 dollar bills
		case bill == 100 && wallet50 > 0:
			wallet100++
			wallet50--
			wallet25--
		case bill == 100 && wallet50 == 0:
			wallet100++
			wallet25 -= 3
		case bill == 50:
			wallet50++
			wallet25--
		case bill == 25:
			wallet25++
		}
		// Second: check if the change was possible
		if wallet25 < 0 {
			return "NO"
		}
	}
	// ok, the queue was processed without an error
	return "YES"
}

File: unsorted/Vasya_Clerk.cs
using System;

public class Line
    {
        private const string yes = "YES";
        private const string no = "NO";
  
        public static string Tickets(int[] peopleInLine)
        { 
            int box100 = 0;
            int box50 = 0;
            int box25 = 0;
            foreach (int invalue in peopleInLine)
                switch (invalue) {
                case 25:
                    box25++;
                    break;
                case 50:
                    box50++;
                    if (box25 > 0)
                        box25--;
                    else
                        return no;
                    break;
                case 100:
                    box100++;
                    if (box50 > 0 && box25 > 0)
                        {box50--; box25--;}
                    else if (box25 > 2)
                        box25 -= 3;
                    else
                        return no;
                    break;
                default:
                    return no;
              }
          return yes;
        }
    }
